torchtitan/models/llama_vision/vision_encoder.py

```diff
@@ -0,0 +1,552 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+#
+# Llama 2 is licensed under the LLAMA 2 Community License,
+# Copyright (c) Meta Platforms, Inc. All Rights Reserved.
+
+from typing import List, Optional
+
+import torch
+import torch.nn as nn
+from torchtitan.models.llama_vision.position_embeddings import (
+    TiledTokenPositionalEmbedding,
+    TilePositionEmbedding,
+    TokenPositionalEmbedding,
+)
+from torchtitan.models.llama_vision.tanh_gate import TanhGate
+from torchtitan.models.types import ModelArgs
+from torchtitan.modules.attention import Attention
+from torchtitan.modules.feed_forward import FeedForward
+from torchtitan.modules.norms import Fp32LayerNorm
+
+
+class Conv2dModule(torch.nn.Module):
+    """Conv2D Module.
+    This is a equivalent to Conv2D in PyTorch, but has support better flexibility
```

Comment:
nit: I think this can paragraph can be phrased more concisely (even if this is taken from the release, I think it is okay for us to do our own pass on the documentation).

This is like Conv2D in PyTorch except:
- PyTorch Conv2D outputs shape `(*, out_channels, h_out, w_out)`, while this module outputs `(*, h_out * w_out, out_channels)`.
- This assumes a square kernel. This allows us to implement the conv as an unfold -> permute -> linear, where we can column-wise shard the linear.